packages/admin/admin/src/stack/breadcrumbs/StackBreadcrumbs.tsx

```diff
@@ -25,18 +25,19 @@ const StackBreadcrumbsComponent = ({
     const [itemWidths, setItemWidths] = React.useState<number[] | undefined>();
 
     const breadcrumbItems = React.useMemo(() => stackApi?.breadCrumbs ?? [], [stackApi]);
+    const combinedTitlesOfBreadcrumbs = breadcrumbItems.map(({ title }) => title).join("");
 
     React.useEffect(() => {
         setItemWidths(undefined);
-    }, [breadcrumbItems]);
+    }, [breadcrumbItems?.length, combinedTitlesOfBreadcrumbs]);
```

Comment:
The problem may happen due to using two useEffect hooks that both set the item widths. Why can't this be just one?

Reply:
No, the problem only occurred when an interaction with the Stack-API triggered a re-initialization of `stackApi.breadCrumbs`, which then triggered the useEffect's even though the values inside the `stackApi.breadCrumbs` array did not change. 

The first useEffect is needed to reset the `itemWidths` when the breadcrumbs change.  
When `itemWidths` is reset (set to `undefined`), all items are shortly rendered, and the second useEffect is triggered, which re-calculates `itemWidths`. 
When  `itemWidths` is set again, only the items that have enough space are shown, the rest are rendered in the overflow menu. 

Whenever the breadcrumbs change, each useEffect's code (inside the `if`) is only called once. 

I'm not sure how this could be done with only one useEffect, since one thing needs to happen, then the component needs to re-render, and then the other thing needs to happen before the final render.

Reply:
Thanks for the explanation!